crates/uv-resolver/src/lock/export/pylock_toml.rs

```diff
@@ -602,3 +1099,51 @@ where
     };
     serializer.serialize_some(&timestamp)
 }
+
+/// Convert a TOML datetime to a Jiff timestamp.
+fn timestamp_from_toml_datetime<'de, D>(deserializer: D) -> Result<Option<Timestamp>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let Some(datetime) = Option::<toml_edit::Datetime>::deserialize(deserializer)? else {
+        return Ok(None);
+    };
+    let Some(date) = datetime.date else {
+        return Err(serde::de::Error::custom("missing date"));
+    };
+
+    let year = i16::try_from(date.year).map_err(serde::de::Error::custom)?;
+    let month = i8::try_from(date.month).map_err(serde::de::Error::custom)?;
+    let day = i8::try_from(date.day).map_err(serde::de::Error::custom)?;
+    let date = civil::date(year, month, day);
+
+    // If the timezone is omitted, assume UTC.
+    let tz = if let Some(offset) = datetime.offset {
+        match offset {
+            toml_edit::Offset::Z => TimeZone::UTC,
+            toml_edit::Offset::Custom { minutes } => {
+                let hours = i8::try_from(minutes / 60).map_err(serde::de::Error::custom)?;
+                TimeZone::fixed(Offset::constant(hours))
```

Comment:
PEP 751 requires UTC, but if we want to support this path, we should use `from_seconds`, there are half-hour timezone offsets.